ts/components/menuAndSettingsHooks/UseTogglePinConversationHandler.tsx

```diff
@@ -0,0 +1,38 @@
+import { useSelector } from 'react-redux';
+import { useIsProAvailable } from '../../hooks/useIsProAvailable';
+import { useHasPro } from '../../hooks/useHasPro';
+import { ConvoHub } from '../../session/conversations';
+import { useIsPinned } from '../../hooks/useParamSelector';
+import {
+  SessionProInfoVariant,
+  useShowSessionProInfoDialogCbWithVariant,
+} from '../dialog/SessionProInfoModal';
+import { Constants } from '../../session';
+import { getPinnedConversationsCount } from '../../state/selectors/conversations';
+
+export function useTogglePinConversationHandler(id: string) {
+  const conversation = ConvoHub.use().get(id);
+  const isPinned = useIsPinned(id);
+
+  const pinnedConversationsCount = useSelector(getPinnedConversationsCount);
+  const isProAvailable = useIsProAvailable();
+  const hasPro = useHasPro();
+
+  const handleShowProDialog = useShowSessionProInfoDialogCbWithVariant();
+
+  if (
+    isPinned ||
+    !isProAvailable ||
+    hasPro ||
+    pinnedConversationsCount < Constants.CONVERSATION.MAX_PINNED_CONVERSATIONS_STANDARD
+  ) {
+    return () => conversation?.togglePinned();
+  }
```

Comment:
see my comment above, but this should show a confirmation modal (see useShowPinUnpin)

Reply:
ignore this too